src/com/github/Kraken3/AFKPGC/BotDetector.java

```diff
@@ -204,45 +203,46 @@ public synchronized void doDetector() {
 									AFKPGC.debug("Player ", curSuspect.getUUID(), " (", curSuspect.getName(), 
 											") is an extreme lag source with a lag compute of ", ls.getLagCompute());
 								} else {
-									Date currentDate = new Date();
-									long bantime = (long) (frequency / currentTPS * 1000.0);
-									/* Because the ban time is based on real time and the next run
-									 * of this method is based on the tick, the ban time needs to be
-									 * adjusted to the tick. The long form of this would be:
-									 * (20/currentTPS) * (frequency/20) * 1000
-									 * The time is in ms */
 									if (!observationMode) {
-										BanEntry leBan = banList.addBan(curSuspect.getName(),
-												"You were suspected to cause lag and banned for "
-														+ (bantime / 1000) + " seconds",
-														new Date(currentDate.getTime() + bantime), null);
-										Player p = Bukkit.getPlayer(curSuspect.getUUID());
-										if (p != null) {
-											if (BotDetector.kickNearby) {
-												List<Player> nearby = getPlayersWithin(p, BotDetector.kickNearbyRadius);
-
-												for (Player q : nearby) {
-													BanEntry qBan = banList.addBan(q.getName(), leBan.getReason(),
-															new Date(currentDate.getTime() + bantime), null);
-													q.kickPlayer(leBan.getReason());
-													AFKPGC.debug("Player ", q.getUniqueId(), " (", q.getName(),
-															") short banned for ", bantime / 1000, 
-															" seconds for being nearby lag source.");
+										if (!warnedPlayers.contains(curSuspect)) {
+											Player p = Bukkit.getPlayer(curSuspect.getUUID());
+											warnPlayer(p);
+											warnedPlayers.add(curSuspect);
+										}
+										else {
+												//The person has been warned already, check whether he is in the same region.
+											Player p = Bukkit.getPlayer(curSuspect.getUUID());
+											if (p != null) {											
+												boolean found = false;
+												for(Suspect warned:warnedPlayers) {
+													if (curSuspect.equals(warned) && curSuspect.getLocation().distance(warned.getLocation()) 

```

Comment:
Just remembered, this is a critical bug. You have to first check if current location and warned location's world are the same.

For this case I think if the worlds are different they've moved far enough away, so it should be an easy fix. I'll take care of it.